flake.nix

```diff
@@ -1,34 +1,54 @@
 {
   inputs.flake-utils.url = github:numtide/flake-utils;
 
-  nixConfig = {
-    # Needed by callCabal2nix
-    allow-import-from-derivation = true;
-  };
-
   outputs = {
     self,
     nixpkgs,
     flake-utils,
   }:
     with flake-utils.lib;
-      eachSystem [system.x86_64-linux] (system: let
+      eachDefaultSystem (system: let
         pkgs = nixpkgs.legacyPackages.${system};
         hpkgs = pkgs.haskellPackages;
-        smtlib-backends = hpkgs.callCabal2nix "smtlib-backends" ./. {};
-        smtlib-backends-process = hpkgs.callCabal2nix "smtlib-backends-process" ./smtlib-backends-process {inherit smtlib-backends smtlib-backends-tests;};
-        smtlib-backends-tests = hpkgs.callCabal2nix "smtlib-backends-tests" ./smtlib-backends-tests {inherit smtlib-backends;};
-        smtlib-backends-z3 = hpkgs.callCabal2nix "smtlib-backends-z3" ./smtlib-backends-z3 {
+        regen = pkgs.writeShellApplication {
+          name = "regen";
+          ## coreutils for xargs
+          runtimeInputs = with pkgs; [cabal2nix findutils coreutils];
+          text = ''
+            set -o errexit
+            set -o pipefail
+            set -o xtrace
+            ## Nix automatically runs `shellcheck` on this script
+            # shellcheck disable=SC2016
+            find . -name '*.cabal' -print0 | \
+            xargs --null -I{} \
+            sh -c 'cabal2nix "$(dirname {})" > "$(basename --suffix='.cabal' {}).nix"'
```

Comment:
Maybe we could have the `smtlib-backends-*.nix` files be in their corresponding directories instead of the top-level one ?

Reply:
I think it would make a lot of sense indeed. We would only need to replace the `.cabal` suffix by `.nix` and then the files would leave side-by-side which would be pretty nice imo. If the filename is a variable `$file`, `${file%.cabal}.nix` would do the trick (eg. in conjunction about the `find | while read` pattern mentioned in [this other conversation](https://github.com/tweag/smtlib-backends/pull/37/files#r1064462256)).